src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallGlobalOrToolPathCommand.cs

```diff
@@ -226,6 +238,22 @@ private int ExecuteInstallCommand(PackageId packageId)
             return 0;
         }
 
+        private NuGetVersion GetBestMatchNugetVersion(PackageId packageId, VersionRange versionRange, IToolPackageDownloader toolPackageDownloader)
+        {
+            return toolPackageDownloader.GetNuGetVersion(
+                packageLocation: new PackageLocation(nugetConfig: GetConfigFile(), additionalFeeds: _source),
+                packageId: packageId,
+                versionRange: versionRange,
+                verbosity: _verbosity,
+                isGlobalTool: true
+            );
+        }
+
+        private static bool ToolVersionAlreadyInstalled(IToolPackage oldPackageNullable, NuGetVersion nuGetVersion)
+        {
+            return oldPackageNullable != null && (oldPackageNullable.Version.Version == nuGetVersion.Version);
```

Comment:
You shouldn't have to do the first half of this check with how this is currently used, but I'd be a bit concerned going forward without it or nullability enabled, so since nullable doesn't seem to be enabled, I think this is good 👍

Reply:
Yeah, I think the same but without nullability enabled and the variable with "nullable" as suffix, better safe then sorry kk